cmd/sops/main.go

```diff
@@ -286,35 +443,41 @@ func main() {
 	app.Run(os.Args)
 }
 
-func runEditor(path string) error {
-	editor := os.Getenv("EDITOR")
-	var cmd *exec.Cmd
-	if editor == "" {
-		cmd = exec.Command("which", "vim", "nano")
-		out, err := cmd.Output()
+func keyservices(c *cli.Context) (svcs []keyservice.KeyServiceClient) {
+	if c.Bool("enable-local-keyservice") {
+		svcs = append(svcs, keyservice.NewLocalClient())
+	}
+	uris := c.StringSlice("keyservice")
+	for _, uri := range uris {
+		url, err := url.Parse(uri)
 		if err != nil {
-			panic("Could not find any editors")
+			log.Printf("Error parsing keyservice URI %s, skipping", uri)
+			continue
+		}
+		addr := url.Host
+		if url.Scheme == "unix" {
+			addr = url.Path
 		}
-		cmd = exec.Command(strings.Split(string(out), "\n")[0], path)
-	} else {
-		parts, err := shlex.Split(editor)
+		opts := []grpc.DialOption{
+			grpc.WithInsecure(),
```

Comment:
is there a reason we're always using WithInsecure?

Reply:
I think adding support for loading TLS client and server certificates for this would end up requiring significant changes to the config. Currently we just suggest using an SSH tunnel to secure the connection. This is mostly a case of us not needing TLS on the connection yet.

Reply:
OK, as long as it's thoroughly documented that this is the case it's ok.

Reply:
I've added a warning for this to the readme: https://github.com/mozilla/sops/pull/238/files#diff-88b99bb28683bd5b7e3a204826ead112R455